src/web/services/table-comparator.service.ts

```diff
@@ -11,6 +12,24 @@ export class TableComparatorService {
 
   constructor() { }
 
+  /**
+   * Compares two strings which are expected to be dates depending on order given.
+   * Assumes that the strings are formatted in the format 'ddd, DD MMM YYYY hh:mm:ss A'.
+   */
+  compareChronologically(strA: string, strB: string, order: SortOrder): number {
+    const dateFormat: string = 'ddd, DD MMM YYYY hh:mm:ss A';
+    const dateA: moment.Moment = moment(strA, dateFormat);
+    const dateB: moment.Moment = moment(strB, dateFormat);
+
+    if (order === SortOrder.ASC) {
+      return dateA.isAfter(dateB) ? 1 : (dateA.isBefore(dateB) ? -1 : 0);
+    }
+    if (order === SortOrder.DESC) {
+      return dateB.isAfter(dateA) ? 1 : (dateB.isBefore(dateA) ? -1 : 0);
+    }
+    return 0;
```

Comment:
How about a `switch` statement?
Also, things look like there could be some refactoring to make it less messy?

Would abstracting the return statement into a util function and only specify the order of `dateA` and `dateB` in the input params highlight what's important here?

Then again, it seems that moment.js provides interface like `isAfter`. Perhaps you can make use of it?

Reply:
To add on, it could probably be simpler if we made use of timestamps, and took the Math.sign (if necessary) of one minus the other: I'm aware of `Date.parse()`, but other moment.js approaches could work too

Reply:
Changes made